libvast/src/system/meta_index.cpp

```diff
@@ -405,8 +405,17 @@ meta_index(meta_index_actor::stateful_pointer<meta_index_state> self,
       self->state.erase(old_partition);
       return atom::ok_v;
     },
+    [=](atom::candidates, vast::expression expr,
+        vast::ids ids) -> caf::result<meta_index_result> {
+      auto query = vast::query{};
+      query.expr = std::move(expr);
+      query.ids = std::move(ids);
+      query.id = vast::uuid::nil();
```

Comment:
If I understand this handler correctly then all it does is to reply to an anonymous query, i.e., a query without an ID. But doesn't it mess up statistics and metrics if we allow that? Why can't we require all queries to have an ID, and simply assign that ID in the constructor of `query` at random?

Reply:
I wasn't sure which way would mess up statistics collection more: if we assign a random id then we'd have lots of queries showing up *only* in the meta index but never receive any follow-up action, and it wouldn't be obvious how to filter them. This way they're all combined with a single id and can be ignored.

Reply:
The difference between regular queries and transforms is that the latter is designed to operate on whole partitions (or at least all events of a certain type therein). We initially decided to keep transforms apart from the `query` to reflect that discrepancy at the type level.

I'm not sure this was the correct decision, as there is definitely some value to having an ID to attach to transform metrics.
In any case, I would probably include the type of operation into the metrics data, irregardless of it being a query or not.
I essentially see 2 alternatives here: 
1. Add a `query::transform` as query command.
2. Add an explicit `uuid id` parameter to this and related handlers.
I slightly prefer (2) in this case.

Also some feedback about code structure:
I would move the logic from the `candidates` handler into a function or method on the actor state and pass the contents of a `query` in destructured form to avoid the ad-hoc query creation here.

Reply:
I refactored the code so that the handler now takes an `id` parameter und uses that to provide metrics.